### PR TITLE
Removed on Roll feature

### PR DESCRIPTION
**Pull request by divios**
Option to enable removeOnRoll, which consist on removing the rolled object when calling roll() so is neved rolled again. I came to this problem in one of my projects, where i wanted a weightedRandom object from a collection but to never appear again. This could be also implemented from the api-user storing the rolled items and if the roll() output is contained, roll again, but i found this to be a better solution since my collection is not really that big.

To sum it up is not a big feature but i wanted to collaborate on the project since i found it really useful on my projects, thanks for all man!

### CODE REVIEW

**Comment by boxbeam** [comment]
I'm hesitant to accept this PR because it's a very narrow use case and can easily be accomplished without the flag you are requesting. There's already WeightedRandom#remove(T), so you could do:

T value = random.roll();
random.remove(value);

You wrote a method called reMap which seems to do the same thing as remove, not sure why. For these reasons I don't think I can accept this PR, sorry.

**Comment by divios** [comment]
Just realized that there is a remove method... Im retarded hahaha and no problem, just wanted to help somehow, have a nice day

**Comment by boxbeam** [comment]
No worries! You opening this PR made me take another look at that method and realize it's inefficient - O(n), so I'm going to optimize it to O(log n) tonight.
